### PR TITLE
gce: Update default instance types

### PR DESCRIPTION
**Pull request by hakman**
These instance types seems newer and with better availability and cost/performance ratio (same type for masters and nodes).

/cc @justinsb 
/assign @justinsb

### CODE REVIEW

**Comment by hakman** [comment]
Added release notes for the changes as discussed during office hours.
/cc @johngmyers @rifelpet

File: docs/releases/1.26-NOTES.md [hunk 1]
```diff
@@ -8,10 +8,10 @@ This is a document to gather the release notes prior to the release.
 
 * The default networking provider for new clusters is now Cilium.
 
-* Control-plane instance groups in new clusters are now created with names starting
+* Control plane instance groups in new clusters are now created with names starting
```
Comment (johngmyers) [inline docs/releases/1.26-NOTES.md:11]:
```suggestion
* Control-plane instance groups in new clusters are now created with names starting
```

Hyphens should be used with compound modifiers.

File: docs/releases/1.26-NOTES.md [hunk 2]
```diff
@@ -27,6 +27,10 @@ with "control-plane-". The names of groups for existing clusters are unchanged.
 
 * New clusters can more easily be configured to use Cilium in ENI mode by setting `--networking=cilium-eni`.
 
+## GCP
+
+* The default instance type is now `e2-medium` for control plane and nodes, and `e2-micro` for bastions.
```
Comment (johngmyers) [inline docs/releases/1.26-NOTES.md:32]:
```suggestion
* The default instance type is now `e2-medium` for control-plane and worker nodes, and `e2-micro` for bastions.
```

**Review by johngmyers — CHANGES_REQUESTED** [review]

File: docs/releases/1.26-NOTES.md (hunk 1 above)
Comment (hakman) [reply docs/releases/1.26-NOTES.md:11]: I was looking at https://kubernetes.io/docs/concepts/architecture/control-plane-node-communication, where they spell it "control plane".

Comment (johngmyers) [reply docs/releases/1.26-NOTES.md:11]: That document doesn't use the term as a compound modifier. When it is used as a compound modifier, English grammar requires it be hyphenated.

File: docs/releases/1.26-NOTES.md (hunk 2 above)
Comment (hakman) [reply docs/releases/1.26-NOTES.md:32]: I can't see that much usage of `control-plane` spelling on https://kubernetes.io.

Comment (johngmyers) [reply docs/releases/1.26-NOTES.md]: Hyphenation of compound modifiers is one of the lesser-well-known English grammar rules.

**Review by johngmyers — APPROVED** [review]

**Comment by johngmyers** [comment]
/hold cancel
I do not understand why Tide tries to merge PRs together when that would make things slower.

**Pull request merged**